aws-cpp-sdk-transfer/include/aws/transfer/UploadFileRequest.h

```diff
@@ -126,6 +126,11 @@ class AWS_TRANSFER_API UploadFileRequest : public S3FileRequest, public std::ena
     bool HasPassedGetObject() const { return m_getObjectPassed.load(); }
     bool HasPassedListObjects() const { return m_listObjectsPassed.load(); }
 
+    // A map of metadata to store with the object in S3.
```

Comment:
I'd like for the map to be in the function call instead of in this object. This object does not contain any other s3 object specific data; it is a handle for callbacks and status checks etc...

Also, when you do so, be sure to handle r-values instead of just a copy.